.github/workflows/release.yml

```diff
@@ -0,0 +1,44 @@
+on:
+  label:
+    types: [created]
+
+# a lot of code taken from https://github.com/pypa/cibuildwheel/blob/main/examples/github-deploy.yml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - run: python -m pip install build
+      - run: python -m build
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: trio-sdist
+          path: dist/*.tar.gz
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: trio-wheel
+          path: dist/*.whl
+
+  pypi-publish:
+    needs: [build]
+    name: upload release to PyPI
+    runs-on: ubuntu-latest
+    environment: release
```

Comment:
1) This refers to a GitHub Deployment Environment — a concept corresponding to some deployment target, like PyPI or TestPyPI. So I always recommend naming it `pypi` in all my guides. Plus William and I are trying to correct this in other places like GitHub's own docs.
2) It's possible to show nice URLs in places where GH shows deployment-related stuff. I like using full versioned URLs when there's an easy way to retrieve the version. But at least something like this would do:
```suggestion
    environment:
      name: pypi
      url: https://pypi.org/p/trio
```

Reply:
Unfortunately I can't seem to change the name for the already created environment. (In fact I wasn't the one to create it, so I don't want to recreate it in fear that it has some special configuration I'll forget.)

I'll add the URL though